python-json-rbac/core.py

```
@@ -96,71 +141,251 @@ def create_token(
+    # Add key ID for rotation tracking if using HS256
+    if ALGORITHM == "HS256":
+        claims["kid"] = key_id or JWT_SECRET_ID
 
     signing_key = _get_signing_key()
-    jwt_token = jwt.encode(claims, signing_key, algorithm=ALGORITHM)
+    
+    try:
+        jwt_token = jwt.encode(claims, signing_key, algorithm=ALGORITHM)
+    except Exception as e:
+        logger.error(f"Failed to create JWT token: {e}")
+        raise RuntimeError(f"Token creation failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)